include/mc_tasks/ImpedanceTask.h

```diff
@@ -0,0 +1,359 @@
+  }
+
+  /*! \brief Set the rotational impedance parameters in the surface. */
+  void impedanceOrientation(const Eigen::Vector3d & impMOri,
+                            const Eigen::Vector3d & impDOri,
+                            const Eigen::Vector3d & impKOri)
+  {
+    impM_.moment() = mc_filter::utils::clampAndWarn(impMOri, impMOriLower_, impMOriUpper_, "impedanceOrientationM");
+    impD_.moment() = mc_filter::utils::clampAndWarn(impDOri, impDOriLower_, impDOriUpper_, "impedanceOrientationD");
+    impK_.moment() = mc_filter::utils::clampAndWarn(impKOri, impKOriLower_, impKOriUpper_, "impedanceOrientationK");
+  }
+
+  /*! \brief Get the impedance mass parameter represented in the surface. */
+  const sva::ForceVecd & impedanceM() const noexcept
+  {
+    return impM_;
+  }
+
+  /*! \brief Set the impedance mass parameter represented in the surface. */
+  void impedanceM(const sva::ForceVecd & impM)
+  {
+    impM_.force() = mc_filter::utils::clampAndWarn(impM.force(), impMPosLower_, impMPosUpper_, "impedancePositionM");
+    impM_.moment() =
+        mc_filter::utils::clampAndWarn(impM.moment(), impMOriLower_, impMOriUpper_, "impedanceOrientationM");
+  }
+
+  /*! \brief Get the impedance damper parameter represented in the surface. */
+  const sva::ForceVecd & impedanceD() const noexcept
+  {
+    return impD_;
+  }
+
+  /*! \brief Set the impedance damper parameter represented in the surface. */
+  void impedanceD(const sva::ForceVecd & impD)
+  {
+    impD_.force() = mc_filter::utils::clampAndWarn(impD.force(), impDPosLower_, impDPosUpper_, "impedancePositionD");
+    impD_.moment() =
+        mc_filter::utils::clampAndWarn(impD.moment(), impDOriLower_, impDOriUpper_, "impedanceOrientationD");
+  }
+
+  /*! \brief Get the impedance spring parameter represented in the surface. */
+  const sva::ForceVecd & impedanceK() const noexcept
+  {
+    return impK_;
+  }
+
+  /*! \brief Set the impedance spring parameter represented in the surface. */
+  void impedanceK(const sva::ForceVecd & impK)
+  {
+    impK_.force() = mc_filter::utils::clampAndWarn(impK.force(), impKPosLower_, impKPosUpper_, "impedancePositionK");
+    impK_.moment() =
+        mc_filter::utils::clampAndWarn(impK.moment(), impKOriLower_, impKOriUpper_, "impedanceOrientationK");
+  }
+
+  /*! \brief Get the wrench gain, which is multiplied by the wrench in the surface frame. */
+  const sva::MotionVecd & wrenchGain() const noexcept
+  {
+    return wrenchGain_;
+  }
+
+  /*! \brief Set the wrench gain, which is multiplied by the wrench in the surface frame. */
+  void wrenchGain(const sva::MotionVecd & gain)
+  {
+    wrenchGain_.linear() =
+        mc_filter::utils::clampAndWarn(gain.linear(), wrenchGainLinLower_, wrenchGainLinUpper_, "wrenchGainLinear");
+    wrenchGain_.angular() =
+        mc_filter::utils::clampAndWarn(gain.angular(), wrenchGainAngLower_, wrenchGainAngUpper_, "wrenchGainAngular");
+  }
```

Comment:
How about adding an `mc_tasks::force::ImpedanceGains` class to lighten the `mc_tasks::force::AdmittanceTask` API?

Something like (simplified for the review):
```cpp
struct ImpedanceGains
{
  /* relevant accessors */
protected:
  // Either sva::ImpedanceVecd or an internal wrapper that ensures the gains are (strictly) positive
  sva::ImpedanceVecd M_;
  sva::ImpedanceVecd D_;
  sva::ImpedanceVecd K_;
  sva::ImpedanceVecd wrench_;
};
```

```cpp
mc_tasks::force::AdmittanceTask task;
// For example, instead of:
task.impedanceK(...);
// We get
task.gains().K(...);
// And with a nice wrapper for ensuring the provided gains are positive we could have:
task.gains().K().linear(...);
// Without having to add as many functions in ImpedanceTask itself
```

Down the line, if we factorize the other force tasks as suggested we can specialized the `ImpedanceGains` accordingly and expose a specific `gains()` method for each force task rather than painstakingly choosing and picking specific methods